### PR TITLE
[Synthetics] Add support for icmp tests

### LINKED ISSUE #1025: Add support for 'icmp' subtype in resource 'datadog_synthetics_test'
### Terraform Version

Terraform v0.14.9

### Affected Resource(s)

-   datadog_synthetics_test

### Terraform Configuration Files

```hcl
resource "datadog_synthetics_test" "icmp-test" {
  type = "api"
  subtype = "icmp"
```

### Expected Behavior

The provider should have option to create an icmp test

### Actual Behavior

```Error: invalid value 'icmp' for SyntheticsTestDetailsSubType: valid values are [http ssl tcp dns multi]```

### PR DESCRIPTION
**Pull request by romainberger**
Add support for icmp Synthetics tests.
Fixes #1025

### CODE REVIEW

File: datadog/tests/resource_datadog_synthetics_test_test.go [hunk 1]
```diff
@@ -332,6 +332,22 @@ func TestAccDatadogSyntheticsDNSTest_Updated(t *testing.T) {
...
+	ctx, accProviders := testAccProviders(context.Background(), t)
+	accProvider := testAccProvider(t, accProviders)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    accProviders,
+		CheckDestroy: testSyntheticsTestIsDestroyed(accProvider),
+		Steps: []resource.TestStep{
+			createSyntheticsICMPTestStep(ctx, accProvider, t),
+			// updateSyntheticsICMPTestStep(ctx, accProvider, t),
```
Comment (jdrouet) [inline datadog/tests/resource_datadog_synthetics_test_test.go:346]: You can just remove the line

**Review by zippolyte — COMMENTED** [review]
LGTM, let's just add `icmp` in the list of possible values in the description of the subtype attribute in the schema def line 40

**Comment by zippolyte** [comment]
You'll need to regenerate the docs too I think with `make docs`

**Comment by romainberger** [comment]
Sorry I always forget to regenerate the docs.

I tried to base the PR on https://github.com/DataDog/terraform-provider-datadog/pull/1041 to be able to update the go client but it's doing some weird stuff, sorry 😕

**Comment by zippolyte** [comment]
I fixed up your branch, let's see what happens

File: datadog/resource_datadog_synthetics_test_.go [hunk 2]
```diff
@@ -181,6 +181,24 @@ func syntheticsTestRequest() *schema.Resource {
 				Type:        schema.TypeBool,
 				Optional:    true,
 			},
+			"number_of_packets": {
+				Description: "Number of pings to use per test for ICMP tests (`subtype = \"icmp\"`).",
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
```
Comment (zippolyte) [inline datadog/resource_datadog_synthetics_test_.go:188]: Let's directly use https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/helper/validation#IntBetween
And also let's add the allowed range to the description

**Review by zippolyte — APPROVED** [review]

**Comment by ksitnik-tc** [comment]
🙏  Thanks guys

**Pull request merged**